pkg/tfvars/vsphere/vsphere.go

```diff
@@ -131,11 +145,18 @@ func createDatacenterFolderMap(infraID string, failureDomains []vtypes.FailureDo
 			tempFolder.Name = infraID
 			failureDomains[i].Topology.Folder = infraID
 		}
+		folder, err := finder.Folder(context.TODO(), tempFolder.Name)
 
-		key := fmt.Sprintf("%s-%s", tempFolder.Datacenter, tempFolder.Name)
-		folders[key] = tempFolder
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I think `finder.Folder` will return an error when no folder is found so you may need to check for that here